lib/modules/datasource/maven/util.ts

```diff
@@ -369,44 +279,51 @@ export function getMavenUrl(
   );
 }
 
-export async function downloadMavenXml(
+export async function downloadMaven(
   http: Http,
-  pkgUrl: URL,
-): Promise<MavenXml> {
-  const protocol = pkgUrl.protocol;
-
-  if (protocol === 'http:' || protocol === 'https:') {
-    const rawResult = await downloadHttpProtocol(http, pkgUrl);
-    const xmlResult = rawResult.transform(({ isCacheable, data }): MavenXml => {
-      const xml = new XmlDocument(data);
-      return { isCacheable, xml };
-    });
-    return xmlResult.unwrapOr({});
+  url: URL,
+): Promise<MavenFetchResult> {
+  const protocol = url.protocol.slice(0, -1);
```

Comment:
I wouldn't slice here (perf)

Reply:
I'm pretty sure this is just an integer decrement under the hood, but okay